benchmarks/dbscan/dbscan_timpl.hpp

```diff
@@ -267,8 +267,8 @@ bool ArborXBenchmark::run(ArborXBenchmark::Parameters const &params)
   {
     Kokkos::Profiling::ProfilingSection profile_total("ArborX::HDBSCAN::total");
     profile_total.start();
-    auto [parents, heights] = ArborX::Experimental::hdbscan(
-        exec_space, primitives, params.core_min_size);
+    auto dendrogram = ArborX::Experimental::hdbscan(exec_space, primitives,
```

Comment:
Why did you change?

Reply:
Because I now considered returning a dendrogram, and it makes sense.

To be clear, right now I don't know what the right interface is, and how do we want to deal with flat clustering. Returning `Dendrogram` allows us to return something we could call a method on, like `auto labels = dendrogram.computeFlatClustering()`.

Reply:
> Because I now considered returning a dendrogram, and it makes sense.

But what's wrong with a structured binding declaration?

Reply:
There's not thing wrong per say. I feel like returning two arrays makes it a bit unclear what to do with them.

Overall, I think the things are still in flux enough that I do not consider the interface finalized in any way. Either way is fine with me.

Reply:
> This is not returning anything.

I don't understand. What is "this"? `hdbscan` function? I'm really confused about what you are saying.